### PR TITLE
feat(workflow): implement transition validation mode engine (task-175)

### PR DESCRIPTION
**Pull request by jpoley**
## Summary

Implements TransitionValidator class with support for three validation modes as specified in task-175:

- **NONE**: Pass-through validation (always succeeds, no gate required)
- **KEYWORD**: Interactive keyword prompt requiring exact string match for manual approval
- **PULL_REQUEST**: Validates that artifacts exist in a merged GitHub PR via gh CLI

### Core Validation
- `validate()`: Main entry point that checks artifacts exist before enforcing validation mode
- `_validate_none()`: Immediate pass-through for automatic transitions
- `_validate_keyword()`: Interactive prompts with clear UI and case-sensitive matching
- `_validate_pull_request()`: GitHub CLI integration to verify merged PRs

### Robustness
- Emergency override flag (`--skip-validation`) for exceptional circumstances
- Comprehensive artifact validation before mode enforcement
- Detailed logging of all validation decisions for audit trail
- Clear error messages for missing dependencies or configuration issues

### Testing
- 33 comprehensive unit tests covering all validation modes
- Mock-based testing for external dependencies (gh CLI)
- Edge case handling (missing artifacts, cancelled prompts, API errors)
- All tests passing with no regressions in existing test suite

## Implementation Details

- Uses gh CLI for GitHub integration (simpler than REST API, leverages existing authentication)
- Validates last 10 merged PRs for artifact existence
- Supports both search-based PR discovery and direct PR number validation
- All validation decisions logged at INFO/WARNING levels for observability
- Interactive keyword prompts with clear visual formatting

## Test Plan

- [x] Unit tests pass (33/33)
- [x] No regressions in existing test suite
- [x] NONE mode always succeeds
- [x] KEYWORD mode validates exact string match (case-sensitive)
- [x] PULL_REQUEST mode integrates with gh CLI
- [x] Artifact validation enforced before mode check
- [x] Skip validation flag bypasses all gates
- [x] Clear error messages for all failure scenarios

## Related

- Task: task-175 - Implement Transition Validation Mode Engine
- Module: `src/specify_cli/workflow/validation_engine.py`
- Tests: `tests/test_validation_engine.py`
- Exports: Added `TransitionValidator` and `TransitionValidationResult` to workflow module

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/specify_cli/workflow/__init__.py [hunk 1]
```diff
@@ -18,17 +8,10 @@
...
-    format_validation_mode,
-    get_transition_by_name,
-    get_transitions_from_state,
-    parse_validation_mode,
-    validate_transition_schema,
+from specify_cli.workflow.prd_validator import (
+    PRDValidationResult,
+    PRDValidator,
+    validate_prd_for_transition,
 )
```
[the remark this answers is not in the document]
Comment (jpoley) [reply src/specify_cli/workflow/__init__.py]: fix this

File: src/specify_cli/workflow/validation_engine.py [hunk 2]
```diff
@@ -0,0 +1,573 @@
...
+from pathlib import Path
+from typing import Any
+
+from specify_cli.workflow.transition import Artifact, TransitionSchema, ValidationMode
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class ValidationResult:
```
[the remark this answers is not in the document]
Comment (jpoley) [reply src/specify_cli/workflow/validation_engine.py]: fix this

File: src/specify_cli/workflow/validation_engine.py [hunk 3]
```diff
@@ -0,0 +1,573 @@
...
+            # Query GitHub for merged PRs related to this feature
+            result = subprocess.run(
+                [
+                    "gh",
+                    "pr",
+                    "list",
+                    "--state",
+                    "merged",
+                    "--search",
+                    f"{feature} in:title",
```
[the remark this answers is not in the document]
Comment (jpoley) [reply src/specify_cli/workflow/validation_engine.py]: fix this

**Comment by jpoley** [comment]
Closing to create clean PR from scratch
